src/shell/scripts/omp.zsh

```diff
@@ -107,30 +111,30 @@ function _posh-tooltip() {
 }
 
 function _posh-zle-line-init() {
-    [[ $CONTEXT == start ]] || return 0
+  [[ $CONTEXT == start ]] || return 0
 
-    # Start regular line editor
-    (( $+zle_bracketed_paste )) && print -r -n - $zle_bracketed_paste[1]
-    zle .recursive-edit
-    local -i ret=$?
-    (( $+zle_bracketed_paste )) && print -r -n - $zle_bracketed_paste[2]
+  # Start regular line editor.
+  (( $+zle_bracketed_paste )) && print -r -n - $zle_bracketed_paste[1]
+  zle .recursive-edit
+  local -i ret=$?
+  (( $+zle_bracketed_paste )) && print -r -n - $zle_bracketed_paste[2]
 
-    eval "$(::OMP:: print transient --status="$omp_last_error" --execution-time="$omp_elapsed" --stack-count="$omp_stack_count" --config="$POSH_THEME" --eval --shell=zsh --shell-version="$ZSH_VERSION" --no-status="$no_exit_code")"
-    zle .reset-prompt
+  eval "$(::OMP:: print transient --status="$omp_last_error" --execution-time="$omp_elapsed" --stack-count="$omp_stack_count" --config="$POSH_THEME" --eval --shell=zsh --shell-version="$ZSH_VERSION" --no-status="$no_exit_code")"
+  zle .reset-prompt
 
-    # If we received EOT, we exit the shell
-    if [[ $ret == 0 && $KEYS == $'\4' ]]; then
-        exit
-    fi
+  # Exit the shell if we receive EOT.
+  if [[ $ret == 0 && $KEYS == $'\4' ]]; then
+    exit
+  fi
 
-    # Ctrl-C
-    if (( ret )); then
-        zle .send-break
-    else
-        # Enter
-        zle .accept-line
-    fi
-    return ret
+  if ((ret)); then
+    # TODO (fix): this is not equal to sending a SIGINT, since the status code ($?) is set to 1 instead of 130.
```

Comment:
I left a comment here because the transient prompt support in Zsh overwrites the built-in `zle-line-init` function, which causes a SIGINT to set `$?` to `1` instead of `130`. I haven't found a fix yet since it is not a fatal issue, IMO. Hope we can resolve this in the future.